### PR TITLE
Bugfix : gatsby-source-wordpress giving invalid node names to GraphQL

### PR DESCRIPTION
**Pull request by sebastienfi**
Should solve https://github.com/gatsbyjs/gatsby/issues/1774

- [x]  Modified regex validating node names.
- [x]  Corrected some issue with the node referencing itself introduced in latest commit.
- [x]  Also prefixed almost every `console.log()` to bring a quieter terminal at build, keeping only necessary info (we still need to see that the process did not hang on heavy sites).

@jquense 
We surely need a better validation of names from GraphQL because `/^[_a-zA-Z][_a-zA-Z0-9]*$/` see as valid some names that will break later on : https://github.com/graphql/graphql-js/pull/632
Example : 
`building schemaName "__jetpack_v4" must not begin with "__", which is reserved by GraphQL introspection. In a future release of graphql this will become a hard error.`

### CODE REVIEW

File: packages/gatsby-source-wordpress/src/gatsby-node.js [hunk 1]
```diff
@@ -68,44 +68,51 @@ exports.sourceNodes = async (
     url = `${_siteURL}/wp-json`
   }
 
-  console.log()
-  console.log(
-    colorized.out(
-      `=START PLUGIN=====================================`,
-      colorized.color.Font.FgBlue
+  if (_verbose) console.log()
```
Comment (jquense) [inline packages/gatsby-source-wordpress/src/gatsby-node.js]: plugins are now passed a `reporter` object for doing this, if you want to make use of it. e.g. reporter.verbose('blah')

File: packages/gatsby-source-wordpress/src/gatsby-node.js [hunk 2]
```diff
@@ -616,8 +631,8 @@ function recursiveAddFields(ent, newEnt) {
 function getValidName(key) {
   let nkey = key
   const NAME_RX = /^[_a-zA-Z][_a-zA-Z0-9]*$/
-  if (!NAME_RX.test(nkey)) {
-    nkey = `_${nkey}`.replace(/-/g, `_`).replace(/:/g, `_`)
+  if (!NAME_RX.test(nkey) || restrictedNodeFields.includes(nkey)) {
+    nkey = `${conflictFieldPrefix}${nkey}`.replace(/-|__|:|\.|\s/g, `_`)
```
Comment (jquense) [inline packages/gatsby-source-wordpress/src/gatsby-node.js]: so we should provide a singfle robust utility for doing this (there is one in gatsby), but this is still incomplete in terms of what graphql will allow. It needs to replace any character that isn't alphanumeric or a `_`. It also can't start with a number or two or more `_`. e.g here are some fields that would still error out here.

- "##foo"
- "1number"

**Comment by corysimmons** [comment]
Is this the problem I'm experiencing in https://github.com/gatsbyjs/gatsby/pull/1321#issuecomment-322117504 ?

File: packages/gatsby-source-wordpress/src/gatsby-node.js (hunk 1 above)
Comment (KyleAMathews) [inline packages/gatsby-source-wordpress/src/gatsby-node.js]: @jquense it'd be cool to auto-prefix the plugin name as well to messages

**Pull request merged**